packages/dev/core/src/Gizmos/planeRotationGizmo.ts

```diff
@@ -122,34 +122,37 @@ export class PlaneRotationGizmo extends Gizmo implements IPlaneRotationGizmo {
             vUV = uv;
         }`;
 
-    protected static _RotationGizmoFragmentShader = `
-        precision highp float;
-        varying vec2 vUV;
-        varying vec3 vPosition;
-        uniform vec3 angles;
-        #define twopi 6.283185307
-        void main(void) {
-            vec2 uv = vUV - vec2(0.5);
-            float angle = atan(uv.y, uv.x) + 3.141592;
-            float delta = gl_FrontFacing ? angles.y : -angles.y;
-            float begin = angles.x - delta * angles.z;
-            float start = (begin < (begin + delta)) ? begin : (begin + delta);
-            float end = (begin > (begin + delta)) ? begin : (begin + delta);
-            float len = sqrt(dot(uv,uv));
-            float opacity = 1. - step(0.5, len);
-
-            float base = abs(floor(start / twopi)) * twopi;
-            start += base;
-            end += base;
-
-            float intensity = 0.;
-            for (int i = 0; i < 5; i++)
-            {
-                intensity += max(step(start, angle) - step(end, angle), 0.);
-                angle += twopi;
+    protected static _createRotationGizmoFragmentShader(color: Color3) {
+        return `
+            precision highp float;
+            varying vec2 vUV;
+            varying vec3 vPosition;
+            uniform vec3 angles;
+            #define twopi 6.283185307
+            void main(void) {
+                vec2 uv = vUV - vec2(0.5);
+                float angle = atan(uv.y, uv.x) + 3.141592;
+                float delta = gl_FrontFacing ? angles.y : -angles.y;
+                float begin = angles.x - delta * angles.z;
+                float start = (begin < (begin + delta)) ? begin : (begin + delta);
+                float end = (begin > (begin + delta)) ? begin : (begin + delta);
+                float len = sqrt(dot(uv,uv));
+                float opacity = 1. - step(0.5, len);
+
+                float base = abs(floor(start / twopi)) * twopi;
+                start += base;
+                end += base;
+
+                float intensity = 0.;
+                for (int i = 0; i < 5; i++)
+                {
+                    intensity += max(step(start, angle) - step(end, angle), 0.);
+                    angle += twopi;
+                }
+                gl_FragColor = vec4(${color.asArray()}, min(intensity * 0.25, 0.8)) * opacity;

```

Comment:
as array generates a new array on every call. To avoid garbage collection it would be better to reference the components directly:
```suggestion
                gl_FragColor = vec4(${color.r}, ${color.g}, ${color.b}, min(intensity * 0.25, 0.8)) * opacity;
```

Reply:
Also, you should make sure that the stringification of r/g/b always has a decimal point, else it will generate an error at shader compilation time. For eg, "1,1,0" won't work, it should be "1.,1.,0." (or "1.0,1.0,0.0").